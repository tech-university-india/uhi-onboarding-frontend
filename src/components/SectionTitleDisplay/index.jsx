import React from 'react'
import Image from 'next/image'
import PropTypes from 'prop-types'

import collapsibleLeft from '../../../public/collapsibleLeft.svg'
import collapsibleDown from '../../../public/collapsibleDown.svg'

/**
 * An expandable section title display. Can also be used without the expandability.
 * Pass the `expanded` prop as well when the `onExpandClick` prop is passed.
 * If the `expanded` prop is not passed, the collapsible arrow is not rendered.
 */
export default function SectionTitleDisplay ({ text, onExpandClick, expanded }) {
  return <div className="p-3 flex flex-row bg-options justify-between items-center border border-gray-100">
    <span className="drop-shadow-none text-2xl font-light">{text}</span>
    <div className="flex flex-row justify-end items-center">
      {expanded !== undefined &&
        <button className="pl-2 bg-transparent" id="collapseBtn" onClick={() => onExpandClick && onExpandClick()}>
          <Image src={expanded ? collapsibleLeft : collapsibleDown} alt={expanded ? 'Collapse' : 'Expand'} />
        </button>
      }
    </div>
  </div>
}

SectionTitleDisplay.propTypes = {
  text: PropTypes.string.isRequired,
  expanded: PropTypes.bool,
  onExpandClick: PropTypes.func
}
