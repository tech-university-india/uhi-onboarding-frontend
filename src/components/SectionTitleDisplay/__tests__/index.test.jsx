import React from 'react'
import SectionTitleDisplay from '..'

import { screen, fireEvent, render } from '@testing-library/react'

describe('SectionTitleDisplay', () => {
  it('should render the component without expandability', () => {
    const dummyPropsNoExpandable = {
      text: 'Section Title'
    }

    const keyValueComponent = render(<SectionTitleDisplay {...dummyPropsNoExpandable} />).asFragment()
    expect(keyValueComponent).toMatchSnapshot()
  })

  it('should render the component with expandability when it is collapsed', () => {
    const dummyPropsCollapsed = {
      text: 'Section Title',
      expanded: false
    }

    const keyValueComponent = render(<SectionTitleDisplay {...dummyPropsCollapsed} />).asFragment()
    expect(keyValueComponent).toMatchSnapshot()
  })

  it('should render the component with expandability when it is expanded', () => {
    const dummyPropsExpanded = {
      text: 'Section Title',
      expanded: true
    }

    const keyValueComponent = render(<SectionTitleDisplay {...dummyPropsExpanded} />).asFragment()
    expect(keyValueComponent).toMatchSnapshot()
  })

  it('should call onExpandClick callback prop when it is collapsed initially', () => {
    const dummyPropsCollapsedWithFn = {
      text: 'Section Title',
      expanded: false,
      onExpandClick: jest.fn()
    }

    expect(dummyPropsCollapsedWithFn.onExpandClick).not.toBeCalled()
    render(<SectionTitleDisplay {...dummyPropsCollapsedWithFn} />)
    const keyValueImage = screen.getByAltText('Expand')
    fireEvent.click(keyValueImage)
    expect(dummyPropsCollapsedWithFn.onExpandClick).toBeCalled()
  })

  it('should call onExpandClick callback prop when it is expanded initially', () => {
    const dummyPropsCollapsedWithFn = {
      text: 'Section Title',
      expanded: true,
      onExpandClick: jest.fn()
    }

    expect(dummyPropsCollapsedWithFn.onExpandClick).not.toBeCalled()
    render(<SectionTitleDisplay {...dummyPropsCollapsedWithFn} />)
    const keyValueImage = screen.getByAltText('Collapse')
    fireEvent.click(keyValueImage)
    expect(dummyPropsCollapsedWithFn.onExpandClick).toBeCalled()
  })
})
