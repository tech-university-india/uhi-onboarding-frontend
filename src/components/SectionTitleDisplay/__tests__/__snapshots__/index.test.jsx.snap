// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionTitleDisplay should render the component with expandability when it is collapsed 1`] = `
<DocumentFragment>
  <div
    class="p-3 flex flex-row bg-options justify-between items-center border border-gray-100"
  >
    <span
      class="drop-shadow-none text-2xl font-light"
    >
      Section Title
    </span>
    <div
      class="flex flex-row justify-end items-center"
    >
      <button
        class="pl-2 bg-transparent"
        id="collapseBtn"
      >
        <img
          alt="Expand"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`SectionTitleDisplay should render the component with expandability when it is expanded 1`] = `
<DocumentFragment>
  <div
    class="p-3 flex flex-row bg-options justify-between items-center border border-gray-100"
  >
    <span
      class="drop-shadow-none text-2xl font-light"
    >
      Section Title
    </span>
    <div
      class="flex flex-row justify-end items-center"
    >
      <button
        class="pl-2 bg-transparent"
        id="collapseBtn"
      >
        <img
          alt="Collapse"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`SectionTitleDisplay should render the component without expandability 1`] = `
<DocumentFragment>
  <div
    class="p-3 flex flex-row bg-options justify-between items-center border border-gray-100"
  >
    <span
      class="drop-shadow-none text-2xl font-light"
    >
      Section Title
    </span>
    <div
      class="flex flex-row justify-end items-center"
    />
  </div>
</DocumentFragment>
`;
